docs/src/modules/components/AppTableOfContents.js

```diff
@@ -201,83 +176,107 @@ class AppTableOfContents extends React.Component {
     }
   };
 
-  handleClick = hash => () => {
+  // Update the active TOC entry if the hash changes through click on '#' icon
+  const handleHashChange = () => {
+    const hash = window.location.hash.substring(1);
+
+    if (activeState !== hash) {
+      setActiveState(hash);
+    }
+  };
+
+  const handleScroll = throttle(() => {
+    findActiveIndex();
+  }, 166); // Corresponds to 10 frames at 60 Hz.
+
+  React.useEffect(() => {
+    itemsClient = getItemsClient(itemsServer);
```

Comment:
`itemsClient` is scoped to the render function of that component. Reassigning it here won't have any effect on the actual render output. Did you mean to put this into state or ref? When refactoring class components instance variables should almost always become refs.
```diff
- this.instanceVar = 5
+ const instanceVarRef = React.createRef(5);
+ const instanceVar = instanceVarRef.current;
```

Reply:
itemsClient only changes when the language is changed, and when that happens, the page reloads (for some reason), so no need to rerender.

Reply:
> itemsClient is scoped to the render function of that component.

Uh, whoops!